sycl/source/detail/global_handler.cpp

```diff
@@ -265,6 +265,7 @@ void GlobalHandler::unloadPlugins() {
       // has been completed to avoid invalid resource access.
       Plugin->call<PiApiKind::piTearDown>(&Plugin->pluginReleased);
       Plugin->unload();
+      Plugin->pluginReleased = true;
```

Comment:
Is there a reason not to do it in `unload()`? Seems like the plugin should register it itself.

Reply:
Good point, it could be done in each unload instead of in the unload plugins. I will update.